src/shared/logger.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as vscode from 'vscode'
+import * as winston from 'winston'
+
+let logSettings: LogSettings
+
+class LogSettings {
+    private readonly outputChannel: vscode.OutputChannel
+    private readonly logger: winston.Logger
+    private readonly logPath: string
+
+    public constructor(outputChannel: vscode.OutputChannel, logPath: string, level: string = 'info') {
+        this.outputChannel = outputChannel
+        this.logger = winston.createLogger({
+            transports: [new winston.transports.File({ filename: logPath })]
+        })
+        this.logPath = logPath
+        // TODO: check if a log level exists--we can add a field for this to the settings page.
+        this.setLevel(level)
+    }
+
+    public getOutputChannel(): vscode.OutputChannel {
+        return this.outputChannel
+    }
+
+    public getLogger(): winston.Logger {
+        return this.logger
+    }
+
+    public setLevel(level: string): void {
+        // TODO: Ensure log level is valid
+        this.logger.level = level
+    }
+
+    public getLevel(): string {
+        return this.logger.level
+    }
+
+    public getLogPath(): string {
+        return this.logPath
+    }
+}
+
+export function initialize(context: vscode.ExtensionContext): void {
+    const outputChannel: vscode.OutputChannel = vscode.window.createOutputChannel('AWS Toolkit Logs')
+    const logPath = context.logPath + '.log'
```

Comment:
Is `context.logPath` a file or directory?

Reply:
Unclear. `context.logPath` doesn't have a file extension, but also doesn't create a directory. I would say that this solution works unless we want to add additional log files

Reply:
Would it make sense to validate that `context.logPath` does not end with `path.sep`, to protect against future changes to the VS Code API implementation?:

Reply:
The [docs](https://code.visualstudio.com/api/references/vscode-api) say:

> An absolute file path **of a directory** in which the extension can create log files. The directory might not exist on disk and creation is up to the extension. However, the parent directory is guaranteed to be existent.